tutorials/physics/interpolation/physics_interpolation_introduction.rst

```diff
@@ -0,0 +1,232 @@
+TPS, then find it breaks on end users machines when it runs at a different tick
+rate.
+
+This can make quality assurance difficult with hard to reproduce bugs, especially
+in AAA games where problems of this sort can be very costly. This can also be
+problematic for multiplayer games for competitive integrity, as running the game at
+certain tick rates may be more advantageous than others.
+
+Lock the tick rate, but use interpolation to smooth frames in between physics ticks
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This has become one of the most popular approaches to dealing with the problem,
+although it is optional and disabled by default.
+
+We have established that the most desirable physics/game logic arrangement for
+consistency and predictability is a physics tick rate that is fixed at design-time.
+The problem is the discrepancy between the physics position recorded, and where we
+"want" a physics object to be shown on a frame to give smooth motion.
+
+The answer turns out to be simple, but can be a little hard to get your head around
+at first.
+
+Instead of keeping track of just the current position of a physics object in the
+engine, we keep track of *both the current position of the object, and the previous
+position* on the previous physics tick.
+
+Why do we need the previous position *(in fact the entire transform, including
+rotation and scaling)*? By using a little math magic, we can use **interpolation**
+to calculate what the transform of the object would be between those two points, in
+our ideal world of smooth continuous movement.
+
+.. image:: img/fti_graph_interpolated.webp
+
+Linear interpolation
+^^^^^^^^^^^^^^^^^^^^
+
+The simplest way to achieve this is linear interpolation, or lerping, which you may
+have used before.
+
+Let us consider only the position, and a situation where we know that the previous
+physics tick X coordinate was 10 units, and the current physics tick X coordinate
+is 30 units.
+
+.. note:: Although the maths is explained here, you do not have to worry about the
+          details, as this step will be performed for you. Under the hood, Godot
+          may use more complex forms of interpolation, but linear interpolation is
+          the easiest in terms of explanation.
+
+The physics interpolation fraction
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If our physics ticks are happening 10 times per second (for this example), what
+happens if our rendered frame takes place at time 0.12 seconds? We can do some math
+to figure out where the object would be to obtain a smooth motion between the two
+ticks.
+
+First of all, we have to calculate how far through the physics tick we want the
+object to be. If the last physics tick took place at 0.1 seconds, we are 0.02
+seconds *(0.12 - 0.1)* through a tick that we know will take 0.1 seconds (10 ticks
+per second). The fraction through the tick is thus:
+
+.. code-block:: python
```

Comment:
What's with the python codeblocks here? Can we use the GDScript highlighting instead?

(Was the 3.6 version of this page originally written before we had GDScript highlighting?)

Reply:
Probably, I'll update the codeblocks.